### PR TITLE
Fixes juggernaut punch spell and new effects

### PR DESCRIPTION
**Pull request by Garen7**
## About The Pull Request

Fixes the juggernaut punch spell so it actually does the 30 damage and the 5 second knockdown it was supposed to do. Angelic constructs punch spell will no longer pass through cultists. New effects for the angelic juggernaut punch spell and force wall(they're basically the same as the normal ones but blue).

## Why It's Good For The Game

The angelic constructs should be able to hit cultists as that's what they were made to do. I made new effects since the old ones looked very cult like especially the part where a mini nar'sie face pops up.

## Changelog
:cl:
tweak: Angelic juggernaut constructs can now use their punch spell on cultists.
fix: Angelic juggernaut constructs' punch spell will now do the intended effect of 30 damage and a 5 second knockdown.
imageadd: Added new effects for the angelic juggernaut construct's spells
/:cl:

### CODE REVIEW

**Comment by Garen7** [comment]
Oops sorry for the commit spam

File: code/modules/spells/spell_types/construct_spells.dm [hunk 1]
```diff
@@ -317,16 +320,33 @@
...
+/obj/projectile/magic/spell/juggernaut/noncult
+	ignored_factions = null
+
+/obj/effect/proc_holder/spell/targeted/projectile/dumbfire/juggernaut/noncult/purified
+	proj_type = /obj/projectile/magic/spell/juggernaut/noncult/purified
+
+/obj/projectile/magic/spell/juggernaut/noncult/purified
+	icon_state = "purecultfist"
+	impact_effect = /obj/effect/temp_visual/dir_setting/firing_effect/magic
+	splash_effect = /obj/effect/temp_visual/dir_setting/firing_effect/magic
```
Comment (Fikou) [inline code/modules/spells/spell_types/construct_spells.dm:352]: newline

Comment (Garen7) [reply code/modules/spells/spell_types/construct_spells.dm:352]: What do you mean?

**Review by Twaticus — APPROVED** [review]

File: code/modules/spells/spell_types/construct_spells.dm (hunk 1 above)
Comment (AnturK) [reply code/modules/spells/spell_types/construct_spells.dm:352]: Add a newline at the end of file.

File: code/modules/spells/spell_types/construct_spells.dm [hunk 2]
```diff
@@ -317,16 +320,33 @@
 	hitsound = 'sound/weapons/punch3.ogg'
 	trigger_range = 0
 	check_holy = TRUE
+	nodamage = FALSE
 	ignored_factions = list("cult")
 	range = 15
 	speed = 7
+	var/obj/effect/temp_visual/impact_effect = /obj/effect/temp_visual/cult/sac
```
Comment (AnturK) [inline code/modules/spells/spell_types/construct_spells.dm:327]: You don't actually keep instances here so var type is bit misleading.

**Review by AnturK — CHANGES_REQUESTED** [review]
Looks good, just some minor stuff

**Pull request merged**